### DISCUSSION TITLE
What is the scope of a variable?

### DISCUSSION THREAD

**Discussion by pmahony893** [mikefarah/yq#1567] [Q&A] [answered]
I ran the following:
```bash
echo "[1, 2, 3]" | yq '[.[] | . as $v | select($v == 1)]'
```

Naively, I expected the result to be:
```
- 1
```

However, the result was:
```
- 1
- 2
- 3
```

I get the same result if I specify `select($v == 2)` or `select($v == 3)`, but for anything else (e.g. `select($v == 0)`), I get `[]`.

I feel I must be misunderstanding something about how variables work in yq pipelines! Can anyone help?

**Comment by mikefarah** [comment, marked as answer]
Hmm I think this is a bug, related to https://github.com/mikefarah/yq/issues/1566 - I'm looking into it

**Reply by pmahony893** [reply]
Thanks! Really appreciate your responsiveness to issues/questions raised here.
